### PR TITLE
chore(NODE-6270): update dev dependencies and add dependabot config

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

Updates dev dependencies and adds a dependabot config.  This PR doesn't update node-gyp and prebuild because our docker images don't have python3.10 installed, required for the new node-gyp version.  

I did try and install python3.10 for a while in our dockerfile, but ran into issues.  We're still on the current major version of node-gyp, we should be fine with this version for a while.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/bindings.test.ts [hunk 1]
```diff
@@ -405,9 +405,12 @@ describe('MongoCryptConstructor', () => {
 });
 
 describe('MongoCryptContext', () => {
-  const context = new MongoCrypt({
-    kmsProviders: serialize({ aws: {} })
-  }).makeDecryptionContext(serialize({}));
+  let context: MongoCryptContext;
```
Comment (baileympearson) [inline test/bindings.test.ts:408]: unrelated to the deps PR: the `test` action has been segfaulting consistently (not just for this PR, see the past runs: https://github.com/mongodb-js/mongodb-client-encryption/actions/workflows/test.yml).  


I reproduced locally just running this single test in a loop.  Instantiating a new context per-test resolves the issues locally.  it's not a permanent fix, I think we should look into this problem separately and maybe it's a libmongocrypt bug, but I wanted green CI for this and future PRs.

File: .eslintrc.json [hunk 2]
```diff
@@ -68,5 +68,8 @@
         "message": "Do not strictly check typeof undefined (NOTE: currently this rule only detects the usage of 'undefined' string literal so this could be a misfire)"
       }
     ]
+    ,
```
Comment (nbbeeken) [inline .eslintrc.json:71]:
```suggestion
    ],
```

File: .eslintrc.json [hunk 3]
```diff
@@ -68,5 +68,8 @@
...
+    "@typescript-eslint/no-unused-expressions": "off",
+    "@typescript-eslint/no-require-imports": "off"
```
Comment (nbbeeken) [inline .eslintrc.json:73]:
```suggestion
    "@typescript-eslint/no-require-imports": "off"
  },
  "overrides": [
    {
      "files": "test/**/*",
      "rules": {
        "@typescript-eslint/no-unused-expressions": "off"
      }
    }
  ]
```

What do you think about applying this rule only to test code? an "unused expression" is neither a typescript error nor a runtime error, so this is the only catch for that potentially easy mistake.

**Review by nbbeeken — CHANGES_REQUESTED** [review]
Minor fix ups

File: .eslintrc.json (hunk 3 above)
Comment (baileympearson) [reply .eslintrc.json:73]: Neutrally.  Do you want me to?

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**